src/tr2/game/text.c

```diff
@@ -48,7 +48,7 @@ void Text_DrawBorder(
 
 void Text_DrawText(TEXTSTRING *const text)
 {
-    if (text->flags.drawn) {
+    if (text->flags.hide || text->flags.drawn) {
```

Comment:
We can remove the duplicate check at line 121, just beneath this piece:
```c
    if (text->flags.hide || text->glyphs == NULL) {
```

Reply:
Thanks, addressed. The conditions now match TR1's `text.c`.